core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -765,12 +765,38 @@ class WorkspaceService(protected val userInfo: UserInfo,
               )
             }
           }
-          workspaces.map { workspace =>
+
+        // Internal folks may create MCWorkspaces in local WorkspaceManager instances, and those will not
+        // be reachable when running against the dev environment.
+        val reachableWorkspaces = workspaces.filter { workspace =>
+          workspace.workspaceType match {
+            case WorkspaceType.McWorkspace =>
+              try {
+                workspaceManagerDAO.getWorkspace(workspace.workspaceIdAsUUID, userInfo.accessToken)
+                true
+              } catch {
+                case e: ApiException =>
+                  logger.warn(s"MC Workspace ${workspace.name} (${workspace.workspaceIdAsUUID}) does not exist in the current WSM instance. ")
```

Comment:
I think the only ApiException that can be reasonably caught here are 404s, correct? In that case this should check the error code on the exception

Reply:
Good point about checking the error code.